************* Module distutils.command.build_ext
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:234:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0401: Wildcard import distutils.errors (wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:29:4: W0621: Redefining name 'show_compilers' from outer scope (line 28) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:29:4: C0415: Import outside toplevel (distutils.ccompiler.show_compilers) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:33:0: C0103: Class name "build_ext" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:33:0: R0902: Too many instance attributes (22/7) (too-many-instance-attributes)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:131:8: C0415: Import outside toplevel (distutils.sysconfig) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:213:25: E1136: Value 'self.plat_name' is unsubscriptable (unsubscriptable-object)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:276:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:130:4: R0912: Too many branches (33/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:130:4: R0915: Too many statements (81/50) (too-many-statements)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:279:8: C0415: Import outside toplevel (distutils.ccompiler.new_compiler) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:342:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:342:4: R0912: Too many branches (14/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:456:12: C0415: Import outside toplevel (concurrent.futures.ThreadPoolExecutor) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:480:8: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:497:8: R1705: Unnecessary "else" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:605:16: C0103: Variable name "o" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:620:8: R1705: Unnecessary "elif" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:630:12: W0120: Else clause on loop without a break statement (useless-else-on-loop)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:627:16: C0103: Variable name "fn" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:615:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:670:8: R1705: Unnecessary "else" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:680:8: C0415: Import outside toplevel (distutils.sysconfig.get_config_var) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:675:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:685:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:715:12: C0415: Import outside toplevel (distutils._msvccompiler.MSVCCompiler) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:734:12: C0415: Import outside toplevel (distutils.sysconfig.get_config_var) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:146:8: W0201: Attribute 'extensions' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:197:16: W0201: Attribute 'build_temp' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:199:16: W0201: Attribute 'build_temp' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:144:12: W0201: Attribute 'package' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:153:12: W0201: Attribute 'include_dirs' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:155:12: W0201: Attribute 'include_dirs' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:249:12: W0201: Attribute 'define' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:255:12: W0201: Attribute 'undef' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:175:12: W0201: Attribute 'libraries' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:177:12: W0201: Attribute 'library_dirs' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:179:12: W0201: Attribute 'library_dirs' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:182:12: W0201: Attribute 'rpath' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:184:12: W0201: Attribute 'rpath' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:306:8: W0201: Attribute 'compiler' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:258:12: W0201: Attribute 'swig_opts' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:260:12: W0201: Attribute 'swig_opts' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:274:16: W0201: Attribute 'parallel' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:538:8: W0201: Attribute '_built_objects' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsModuleError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsClassError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsGetoptError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsArgError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsFileError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsExecError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsInternalError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsTemplateError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import DistutilsByteCompileError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import PreprocessError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import LibError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import LinkError from wildcard import (unused-wildcard-import)
/log/firmware/firmware/usr/lib/python3.8/distutils/command/build_ext.py:12:0: W0614: Unused import UnknownFileError from wildcard import (unused-wildcard-import)

-----------------------------------
Your code has been rated at 8.18/10

/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:145:8: W0201: Attribute 'shlibs' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:143:8: W0201: Attribute 'extensions' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:207:16: W0201: Attribute 'compiler' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:213:12: W0201: Attribute 'compiler' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:294:4: R0913: Too many arguments (13/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:307:4: C0103: Constant name "libtype" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:309:4: R0913: Too many arguments (13/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:309:4: R0914: Too many local variables (16/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:310:60: W0613: Unused argument 'libraries' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:311:12: W0613: Unused argument 'library_dirs' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:311:31: W0613: Unused argument 'runtime_library_dirs' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:311:58: W0613: Unused argument 'export_symbols' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:312:21: W0613: Unused argument 'extra_preargs' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:312:41: W0613: Unused argument 'extra_postargs' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:312:62: W0613: Unused argument 'build_temp' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:322:18: W0612: Unused variable 'ext' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:33:0: C0411: standard import "from distutils.sysconfig import _config_vars as _CONFIG_VARS" should be placed before "from setuptools.extension import Library" (wrong-import-order)
/log/firmware/firmware/usr/lib/python3.8/site-packages/setuptools/command/build_ext.py:33:0: C0412: Imports from package distutils are not grouped (ungrouped-imports)

-----------------------------------
Your code has been rated at 3.30/10

